cmake/modules/Findcrypto.cmake

```diff
@@ -56,7 +56,10 @@ else()
     )
 
     if (NOT crypto_LIBRARY)
-        if (BUILD_SHARED_LIBS)
+        if(${AWS_CRYPTO_BUILD_SHARED_LIBS})
```

Comment:
Can you document this option, similar to https://github.com/aws/aws-sdk-cpp/pull/2829/files#diff-1e7de1ae2d059d21e1dd75d5812d5a34b0222cef273b7c3a2af62eb747f9d20aR69

Reply:
yep sounds good, can add it [here](https://github.com/aws/s2n-tls/blob/main/CMakeLists.txt#L36), since it is your option, but we are using it, are you fine with the name `AWS_CRYPTO_BUILD_SHARED_LIBS` or would you like `S2N_CRYPTO_BUILD_SHARED_LIBS`? 

from a SDK perspective we want it this way so that a user only has to specify one cmake option, but if you feel differently, im ok with that.

Reply:
I'd prefer `S2N_BUILD_SHARED_LIBS` so that future combinations of static/dynamic linking don't require any more options to be added.

Also, thoughts on matching the current behavior? E.g.
```
if (BUILD_SHARED_LIBS || S2N_BUILD_SHARED_LIBS)
...
```
That feels conceptually simpler to me.

Reply:
yeah thats makes sense to me, updated